src/lib/time.js

```diff
@@ -219,7 +215,7 @@ var $builtinmodule = function (name) {
     mod.asctime = new Sk.builtin.func(asctime_f);
 
     mod.ctime = new Sk.builtin.func(function(secs) {
-        return asctime_f(localtime_f(secs));
+        return asctime_f(from_seconds(secs));
```

Comment:
I think you've lost the arg lengths check in this function while doing the refactor

Reply:
Maybe I'm misunderstanding, but the original call, `localtime_f(secs)`, passes exactly one argument to `localtime_f()` and so won't the arg lengths check in the original `localtime_f()` always pass, no matter what was passed to `ctime()`?  Testing against `master`, i.e., *without* this PR, seems to confirm this:
```python
print(time.ctime())  # -> "Thu Nov 26 14:46:35 2020" as expected
print(time.ctime(1606401981))  # -> "Thu Nov 26 14:46:21 2020" as expected
print(time.ctime(1606401981, 1, 1, 1))  # -> "Thu Nov 26 14:46:21 2020" ?? should be error
```
I can add args length checks to `ctime()` and `asctime()` if that would be useful?

Reply:
ah yes - so it was a bug already - can you address this as part of the refactor
in cyptyon
```python
>>> time.ctime(1606401981, 1, 1, 1)
TypeError: ctime() takes at most 1 argument (4 given)
```

Reply:
I've rebased onto `master`, and added a couple of commits which test that the appropriate exception is raised when too many arguments (or not enough) are passed.  Some of these tests initially fail (`asctime`, `ctime`, `mktime`); these are fixed by the newly-last commit in this PR.